### PR TITLE
Add request weather and show result

### PR DESCRIPTION
**Pull request by AndrewHelix**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1 @@
+TOKEN = eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwOlwvXC9wZmEuZm9yZWNhLmNvbVwvYXV0aG9yaXplXC90b2tlbiIsImlhdCI6MTY2MTI1NDYyOSwiZXhwIjo5OTk5OTk5OTk5LCJuYmYiOjE2NjEyNTQ2MjksImp0aSI6ImU0OGVlZmRjZTQwZTI0ZmEiLCJzdWIiOiJhbmRyZXlzaHVsZ2E5NSIsImZtdCI6IlhEY09oakM0MCtBTGpsWVR0amJPaUE9PSJ9.B6s3P_fv-RgW5v2S2x0JD8fhxar1L1fSRn5iTUh1-g0
```
Comment (JuliaShavyrova) [inline .env]: no hardcoded tokens please :)

File: src/components/main/Main.tsx [hunk 2]
```diff
@@ -1,26 +1,67 @@
...
+  });
+
+  async function fetchData(lon: number, lat: number) {
+    await getCity(lon, lat).then(locationData => setCity(locationData.city));
+    await getCityWeather(lon, lat).then(cityWeather => setWeather(cityWeather));
+  }
+
+  useEffect(() => {
+    if (position.longitude) {
+      const lon = position.longitude as number;
```
Comment (JuliaShavyrova) [inline src/components/main/Main.tsx]: `as` should be used only in exceptional cases when there is no way to define typings directly (when 3rd party libraries used, for example) so please refactor the code

File: src/components/main/Main.tsx [hunk 3]
```diff
@@ -1,26 +1,67 @@
...
+      const lat = position.latitude as number;
+
+      fetchData(lon, lat);
+    }
+  }, [position.longitude]);
+
+  useEffect(() => {
+    console.log(weather);
+  }, [weather]);
```
Comment (JuliaShavyrova) [inline src/components/main/Main.tsx]: please remove console log

File: src/services/httpService.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+async function HTTPRequest(
+  endpoint = '',
+  { method = 'GET', body = null, headers = { 'Content-Type': 'application/json' } }
+) {
+  const request = await fetch(`${URL}${endpoint}`, {
+    method,
+    headers: {
+      authorization: `Bearer ${TOKEN}`,
+      ...headers
```
Comment (JuliaShavyrova) [inline src/services/httpService.ts:13]: why not add content type header here?

File: .env [hunk 5]
```diff
@@ -0,0 +1,2 @@
+USER = andreyshulga95
+PASSWORD = 2Eqlsa9xcN90
```
Comment (JuliaShavyrova) [inline .env]: now i can use your creds, thanks :)

File: src/components/app/App.tsx [hunk 6]
```diff
@@ -1,30 +1,75 @@
...
 const App: FC = () => {
+  const [token, setToken] = useState<string>('');
+
+  async function getToken(user: string, password: string): Promise<string> {
+    const body = {
+      user,
+      password
+    };
+
+    const ENV = process.env.NODE_ENV === 'production' ? 'production' : 'development';
```
Comment (JuliaShavyrova) [inline src/components/app/App.tsx]: variable can be defined outside the component body

File: src/components/app/App.tsx [hunk 7]
```diff
@@ -1,30 +1,75 @@
...
+
+    let URL =
+      ENV === 'development'
+        ? 'https://cors.bridged.cc/https://pfa.foreca.com/authorize/token?expire_hours=-1'
```
Comment (JuliaShavyrova) [inline src/components/app/App.tsx]: please don't use 3rd party services as proxy

File: src/services/httpService.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
+import { getFromLocalStorage } from './localStorage';
+
+const URL = 'https://pfa.foreca.com';
+
+async function HTTPRequest(endpoint = '', { method = 'GET', body = '', headers = {} }) {
+  const TOKEN = getFromLocalStorage('token');
```
Comment (JuliaShavyrova) [inline src/services/httpService.ts]: please use camelCase for simple variables

File: src/services/httpService.ts [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+  const request = await fetch(`${URL}${endpoint}`, {
+    method,
+    headers: {
+      'Content-Type': 'application/json',
+      authorization: `Bearer ${TOKEN}`,
+      ...headers
+    },
+    body: body ? body : null
```
Comment (JuliaShavyrova) [inline src/services/httpService.ts]:
```suggestion
    body: body || null
```

File: .env (hunk 5 above)
Comment (AndrewHelix) [reply .env]: you mean I need to add a file .env in .gitignore?

File: src/components/app/App.tsx (hunk 7 above)
Comment (AndrewHelix) [reply src/components/app/App.tsx]: Write my own server for resolve CORS mistake?

File: .env (hunk 5 above)
Comment (JuliaShavyrova) [reply .env]: yes

File: src/components/app/App.tsx (hunk 7 above)
Comment (JuliaShavyrova) [reply src/components/app/App.tsx]: yes

File: src/proxyServer/index.js [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+  return res;
+}
+
+app.get('/token', (_, res) => {
+  getToken()
+    .then(result => res.json(result))
+    .catch(err => console.log(err));
+});
+
+app.listen(8081, '127.0.0.1', () => {
```
Comment (JuliaShavyrova) [inline src/proxyServer/index.js]: why do you need to define '127.0.0.1'?

File: src/components/app/App.tsx [hunk 11]
```diff
@@ -1,30 +1,55 @@
...
 import { Details } from 'components/details/Details';
 import { Feedback } from 'components/feedback/Feedback';
 import { Page404 } from 'components/page404/Page404';
+import { setInLocalStorage, getFromLocalStorage } from 'services/localStorage';
 
 const App: FC = () => {
+  const [token, setToken] = useState<string>('');
+
+  async function getToken(): Promise<string> {
+    const result = await fetch('http://127.0.0.1:8081/token');
```
Comment (JuliaShavyrova) [inline src/components/app/App.tsx]: does it work now?

Comment (AndrewHelix) [reply src/components/app/App.tsx]: fixed it and it worked
localhost = 127.0.0.1 or no?

Comment (JuliaShavyrova) [reply src/components/app/App.tsx]: Yes it is, but it might be a bit confusing :)

**Review by JuliaShavyrova — APPROVED** [review]

**Pull request merged**